### PR TITLE
Add support for UCF-101 dataset and upgrade ffmpeg  version from 3.4.2 to 4.2

### PR DESCRIPTION
**Pull request by a-sansanwal**
#### Why we need this PR?
- Refactoring to improve *what*
Currently VideoReader did not support reading from UCF-101 dataset as pointed out in #1169 

#### What happened in this PR?
 - Explain solution of the problem, new feature added.
We enable avi demuxer, this allows us to demux avi container. We also enable the mpeg4_unpack_bframes bitstream filter and run it on mpeg4/divx codec files that are likely to have packed b-frames.

 - What was changed, added, removed?
Added avi demux support
Enabled mpeg4_unpack_bframes  filter
Ffmpeg upgraded to 4.2 version. This was needed for the av_bsf_flush API which was not present in 3.4.2.

 - What is most important part that reviewers should focus on?
`VideoLoader::get_or_open_file`
 - Was this PR tested? How?
Tested with UCF-101 dataset.
 - Were docs and examples updated, if necessary?
Docs updated to reflect 3.4.2 -> 4.2 upgrade.

We need to add 4.2 tarball to https://developer.download.nvidia.com/compute/redist/nvidia-dali/.

### CODE REVIEW

File: Dockerfile.deps [hunk 1]
```diff
@@ -121,8 +121,8 @@ RUN CLANG_VERSION=6.0.1 && \
 ENV NVIDIA_DRIVER_CAPABILITIES=video,compute,utility
 
 # FFmpeg
-RUN FFMPEG_VERSION=3.4.2 && \
-    cd /tmp && wget https://developer.download.nvidia.com/compute/redist/nvidia-dali/ffmpeg-${FFMPEG_VERSION}.tar.bz2 && \
+RUN FFMPEG_VERSION=4.2 && \
+    cd /tmp && wget https://ffmpeg.org/releases/ffmpeg-${FFMPEG_VERSION}.tar.bz2 && \
```
Comment (Kh4L) [inline Dockerfile.deps]:
```suggestion
    cd /tmp && wget https://developer.download.nvidia.com/compute/redist/nvidia-dali/ffmpeg-${FFMPEG_VERSION}.tar.bz2 && \
```

**Comment by a-sansanwal** [comment]
> I think we should have some kind of simple test for this. We can add artificial file to DALI_extra and try to read it.

I believe we can add a few videos from UCF-101 with packed b-frames to DALI_extra if licensing allows it and a test case for that.
I'm not 100 percent sure but I doubt there are licensing issues because tensorflow has included it. see https://github.com/tensorflow/datasets/pull/409.

**Comment by awolant** [comment]
!build

**Comment by a-sansanwal** [comment]
I dont think the build will pass without ffmpeg 4.2 tarball

**Comment by Remember2018** [comment]
cmake -DFFMPEG_ROOT_DIR="/usr/local/" ../

**Pull request merged**